app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -1,6 +1,7 @@
 module Mixins
   module GenericShowMixin
     def show
+      return if perfmenu_click?
```

Comment:
yes you've got it right, it's for timelines and utilization graphs (clicks), those are displayed in show action so the clicks on graphs are using ```params[:menu_click]``` to distinguish them, and they have different handling as other show actions. You can see it that inside the ```perfmenu_click?``` method is called method ```perf_menu_click``` from https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller/performance.rb#L172 .. I agree that the clicks into those graphs should call different method (not show) as the clicks are not necessary related to the show action. I've just moved it there as I don't like to change it in these refactoring(s) - you can see the same in https://github.com/ManageIQ/manageiq-ui-classic/pull/1037/files#diff-a0489893ffd578c9668e837723389e67L12 and https://github.com/ManageIQ/manageiq-ui-classic/pull/1039/files#diff-c825546e2b4be699311249312a23bd63L18 .. maybe some more of them in near future, so I've put them inside the generic show method for now as we'd like to move forward with these cleanings of the show methods in the controllers. So yea I think it's wise to do any change of that later